### PR TITLE
fix: revert previous facet changes and make new changes

### PR DESCRIPTION
**Pull request by ColbyJohnIBM**
#### What do these changes do/fix?
- revert previous commit (changes were requested by Chris Cook [here](https://ibm-analytics.slack.com/archives/G0117CFBJTU/p1658509360115819?thread_ts=1658332780.760249&cid=G0117CFBJTU))
- re-add double quotes to query
- adjust query decoding to use '::'
- clean up test warnings and logs

Contributes to https://github.ibm.com/Watson-Discovery/disco-issue-tracker/issues/12682

#### How do you test/verify these changes?
Try the following on each of these two [staging](https://us-south.discovery.test.watson.cloud.ibm.com/v2/instances/crn%3Av1%3Astaging%3Apublic%3Adiscovery%3Aus-south%3Aa%2F24f453567f259b67c34ae38a6e6026fb%3A97732895-f1ad-4a2b-9951-548ee712f480%3A%3A/projects/918a1017-e178-4bc8-9b0e-a5950763ac16/workspace) / [local](http://localhost:4000/instances/crn%3Av1%3Astaging%3Apublic%3Adiscovery%3Aus-south%3Aa%2F24f453567f259b67c34ae38a6e6026fb%3A97732895-f1ad-4a2b-9951-548ee712f480%3A%3A/projects/918a1017-e178-4bc8-9b0e-a5950763ac16/workspace) (with this branch and linked to tooling)
1. Run an empty search.
2. Apply a `National Institute of Standards and Technology` facet filter.
3. Note that staging returns nothing, but local returns 56 results.

### CODE REVIEW

File: packages/discovery-react-components/src/components/SearchFacets/utils/searchFilterTransform.tsx [hunk 1]
```diff
@@ -79,50 +65,38 @@ export class SearchFilterTransform {
...
 
-  private static unescapeString(escapedString: string): string {
-    // Remove the parentheses added by escapeSelectedFacets
-    // Unescape the special characters within the string
-    const unescapedString = escapedString
-      .replace(RegExp(SearchFilterTransform.STRING_IN_PARENS), '$1')
-      .replace(RegExp(SearchFilterTransform.ESCAPED_CHAR), '$1');
-    return unescapedString;
+  private static unquoteString(quotedString: string): string {
+    return quotedString.replace(/^"(.+)"$/, '$1').replace(/\\"/, '"');
```
Comment (jhpedemonte) [inline packages/discovery-react-components/src/components/SearchFacets/utils/searchFilterTransform.tsx:75]: I think you should keep change to make this regex a `const` (with comment)

File: packages/discovery-react-components/src/components/SearchFacets/utils/searchFilterTransform.tsx [hunk 2]
```diff
@@ -79,50 +65,38 @@ export class SearchFilterTransform {
...
   static fieldsToString(facets: InternalQueryTermAggregation[]): string {
     const filterStrings: string[] = [];
     facets.forEach(facet => {
       const field = get(facet, 'field', '');
       const results = get(facet, 'results', []);
-      const keys = this.escapeSelectedFacets(results, 'key');
+      const keys = this.quoteSelectedFacets(results, 'key');
       if (keys.length) {
-        filterStrings.push(keys.map(key => `${escapeFieldName(field)}:${key}`).join('|'));
+        filterStrings.push(keys.map(key => `${escapeFieldName(field)}::${key}`).join('|'));
```
Comment (jhpedemonte) [inline packages/discovery-react-components/src/components/SearchFacets/utils/searchFilterTransform.tsx:85]: Same as before: add comment to function, stating that it is using double-colon and what effect(s) that has.

File: packages/discovery-react-components/src/components/SearchFacets/utils/searchFilterTransform.tsx [hunk 3]
```diff
@@ -8,24 +8,10 @@ import {
   SelectableDynamicFacets
 } from './searchFacetInterfaces';
 
-/*
- * Utility class for creating and parsing query strings for search filters
- */
```
Comment (jhpedemonte) [inline packages/discovery-react-components/src/components/SearchFacets/utils/searchFilterTransform.tsx:13]: Keep this

**Review by jhpedemonte — APPROVED** [review]

File: examples/discovery-search-app/cypress/integration/multi_select_facets/spec.ts [hunk 4]
```diff
@@ -48,7 +48,7 @@ describe('Multi-Select Facets', () => {
       it('queries and selects the right facets', () => {
         cy.get('@amesFilterQueryObject')
           .its('requestBody.filter')
-          .should('eq', 'location:(Ames\\, IA)');
+          .should('eq', 'location:"Ames, IA"');
```
Comment (noah-eigenfeld) [inline examples/discovery-search-app/cypress/integration/multi_select_facets/spec.ts:51]: Ha! I forgot that I added this test data with my college town

**Review by noah-eigenfeld — APPROVED** [review]

**Comment by ColbyJohnIBM** [comment]
Unfortunately, this brings back in the LGTM complaining about the old regexes, but I'm not sure how to simplify the regexes to make them cleaner and more maintainable...

**Pull request merged**